### PR TITLE
Oidc groups and refresh

### PR DESCRIPTION
**Pull request by dskatz**
* Adds CI/CD
* This has been refactored to reflect an upstream PR (https://github.com/dexidp/dex/pull/1434). Once this PR is merged we will no longer need to maintain this fork.

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    commands:
+      - make test
+
+  publish_quay:
+    image: plugins/docker
+    secrets: [docker_username, docker_password]
+    registry: quay.io
+    repo: quay.io/mongodb/dex
+    tags:
+      - git-${DRONE_COMMIT_SHA:0:7}
```
Comment (colinhoglund) [inline .drone.yml]: Needs a version tag as well.

File: .drone.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    when:
+      event: push
```
Comment (colinhoglund) [inline .drone.yml]: This is going to run on a push to any branch. You should make the entire pipeline conditional by putting `branches: master` at the bottom.

File: connector/oidc/oidc.go [hunk 3]
```diff
@@ -260,6 +261,17 @@ func (c *oidcConnector) createIdentity(ctx context.Context, identity connector.I
 		}
 	}
 
+	cd := connectorData{
+		RefreshToken: []byte(token.RefreshToken),
+	}
+
+	c.logger.Debugf("refresh token: %v", string(cd.RefreshToken))
```
Comment (colinhoglund) [inline connector/oidc/oidc.go]: I don't think we ever want to log this value.

File: .drone.yml (hunk 1 above)
Comment (dskatz) [reply .drone.yml]: Added a new stage that will support a release tag.

**Pull request merged**